R/app.R

```diff
@@ -496,3 +506,17 @@ knit_print.reactive <- function(x, ..., inline = FALSE) {
     x()
   }), inline = inline)
 }
+
+#' Run code after app has finished
+#'
+#' This function allows you to register a callback function that is invoked
+#' after your app has finished running (all sessions have been disconnected).
+#'
+#' @param fun A function that will be called after the app has finished running.
+#'
+#' @examples
+#' onStop(function() { print("goodbye") })
+#' @export
+onStop <- function(fun) {
+  shinyOptions(onStop = fun)
```

Comment:
@bborgesr As far as I can tell, one principle utility of multiple `onStop` calls is to support the presence of (multiple) libraries that manage resources. Being able to register multiple `onStop` calls makes it possible for users' applications to use multiple libraries, each of which might manage a resource like a database or connection, without the user needing to necessarily install an `onStop` function that calls of the appropriate shutdown/cleanup APIs for each library. The libraries can register to perform this cleanup on their own.